identity/backend/src/main/java/io/camunda/identity/usermanagement/service/UserService.java

```diff
@@ -8,15 +8,19 @@
 package io.camunda.identity.usermanagement.service;
 
 import io.camunda.authentication.user.CamundaUserDetailsManager;
-import io.camunda.identity.user.CamundaUser;
-import io.camunda.identity.user.CamundaUserWithPassword;
+import io.camunda.identity.usermanagement.CamundaUser;
+import io.camunda.identity.usermanagement.CamundaUserWithPassword;
+import io.camunda.identity.usermanagement.model.Profile;
 import io.camunda.identity.usermanagement.repository.UserRepository;
 import java.util.List;
+import java.util.Objects;
 import org.springframework.dao.DuplicateKeyException;
+import org.springframework.security.core.userdetails.User;
 import org.springframework.security.core.userdetails.UserDetails;
 import org.springframework.security.core.userdetails.UsernameNotFoundException;
 import org.springframework.security.crypto.password.PasswordEncoder;
 import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
 
 @Service
 public class UserService {
```

Comment:
I'd suggest to put `@Transaactional` at the class level as it's already on most methods. Would it hurt to do so? For methods that only use one repository afaik the transaction of the service is used for the repo as well, no new transaction will be created.

I used to do this on all services in my past to have the common pattern of service methods being transaction and only make rare exceptions when needed. Would that be an interesting approach for us too maybe?